TrackletAlgorithm/TrackHandler.cc

```diff
@@ -1,82 +1,81 @@
-    auto masterDiskStubIndex = ap_uint<TrackFitType::kTFStubIndexSize>(masterTrk._diskStubArray[diskStubNum][0].range(kDiskStubIndexSizeMSB, kDiskStubIndexSizeLSB));
-    auto inputDiskStubIndex = ap_uint<TrackFitType::kTFStubIndexSize>(trk._diskStubArray[diskStubNum][0].range(kDiskStubIndexSizeMSB, kDiskStubIndexSizeLSB));
-    if ((masterDiskStubIndex == inputDiskStubIndex) && (masterDiskStubIndex > 0)){
-      matchesFoundDisk[diskStubNum][0] = 1;
-    } else {
-      matchesFoundDisk[diskStubNum][0] = 0;
-    }
+    #pragma HLS unroll
+    ap_uint<TrackFitType::kTFStubIndexSize> masterDiskStubIndex = ap_uint<TrackFitType::kTFStubIndexSize>(masterTrk._diskStubArray[diskStubNum][0].range(kDiskStubIndexSizeMSB, kDiskStubIndexSizeLSB));
+    ap_uint<TrackFitType::kTFStubIndexSize> inputDiskStubIndex = ap_uint<TrackFitType::kTFStubIndexSize>(trk._diskStubArray[diskStubNum][0].range(kDiskStubIndexSizeMSB, kDiskStubIndexSizeLSB));
+
+    ((masterDiskStubIndex == inputDiskStubIndex) && (masterDiskStubIndex > 0)) ? matchesFoundDisk[diskStubNum][0] = 1 : matchesFoundDisk[diskStubNum][0] = 0;
+    
   }
 
-  // update #matches found 
+  // Update #matches found 
   unsigned int matchesFound = 0;
   LOOP_MatchesFound:
   for (unsigned int layerIndex = 0; layerIndex < 4; layerIndex++){
     #pragma HLS unroll
     matchesFound += matchesFoundBarrel[layerIndex][0] + matchesFoundDisk[layerIndex][0];
   }
-  if (matchesFound >= mergeCondition){
-    matchFound = 1;
-    return matchFound; 
-    } else {
-      matchFound = 0;
-      return matchFound;
-    }
+  (matchesFound >= mergeCondition) ? matchFound = 1 : matchFound = 0; 
+
+  return matchFound;
 }
 
-void TrackHandler::mergeTrack(TrackStruct& trk, TrackStruct& masterTrk){
-  // update master track
-  // check whether the stub word is non-zero in the compared track
-  // then add stub into master track if share > 3 stubs in common
+void TrackHandler::mergeTrack(const TrackStruct &trk, TrackStruct &masterTrk){
+  #pragma HLS array_partition variable=masterTrk._barrelStubArray complete dim=0
+  #pragma HLS array_partition variable=trk._barrelStubArray complete dim=0
+  #pragma HLS array_partition variable=masterTrk._diskStubArray complete dim=0
+  #pragma HLS array_partition variable=trk._diskStubArray complete dim=0
+  // Update master track
+  // Check whether the stub word is non-zero in the compared track
+  // Then add stub into master track if share > 3 stubs in common
   LOOP_SetInputBarrelStubToMaster:
   for(unsigned int barrelStubNum = 0; barrelStubNum < NBarrelStub; barrelStubNum++){
     #pragma HLS unroll
     if((masterTrk._barrelStubArray[barrelStubNum][0] == 0) && (trk._barrelStubArray[barrelStubNum][0] != 0)){
       masterTrk._barrelStubArray[barrelStubNum][0] = trk._barrelStubArray[barrelStubNum][0];
-      mergedBarrelStubsMap += (1 << TrackFitType::kTFHitCountSize*barrelStubNum);
+      mergedBarrelStubsMap | (1 << TrackFitType::kTFHitCountSize*barrelStubNum);
```

Comment:
@kwalkingshaw This line and the corresponding one for `mergedDiskStubsMap` don't actually do anything. They calculate a bitwise OR, but it isn't stored anywhere…